features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -999,19 +1004,31 @@ nsapi_error_t AT_CellularNetwork::scan_plmn(operList_t &operators, int &opsCount
 }
 
 nsapi_error_t AT_CellularNetwork::set_ciot_optimization_config(Supported_UE_Opt supported_opt,
-                                                               Preferred_UE_Opt preferred_opt)
+                                                               Preferred_UE_Opt preferred_opt,
+                                                               Callback<void(Supported_Network_Opt)> network_support_cb)
 {
+    _ciot_network_support_cb = network_support_cb;
     _at.lock();
 
     _at.cmd_start("AT+CCIOTOPT=");
-    _at.write_int(_cid);
+    _at.write_int(1); //enable CCIOTOPTI URC
     _at.write_int(supported_opt);
     _at.write_int(preferred_opt);
     _at.cmd_stop_read_resp();
 
+
     return _at.unlock_return_error();
 }
 
+void AT_CellularNetwork::urc_cciotopti()
+{
+    Supported_Network_Opt supported_network_opt = (Supported_Network_Opt)_at.read_int();
+
+    if (_ciot_network_support_cb) {
+        _ciot_network_support_cb(supported_network_opt);
+    }
+}
+
 nsapi_error_t AT_CellularNetwork::get_ciot_optimization_config(Supported_UE_Opt &supported_opt,
                                                                Preferred_UE_Opt &preferred_opt)
 {
```

Comment:
This API name should be changed to indicate that this will return **UE** ciot optimizations.

Reply:
I think a getter for the Network's ciot optimization is needed. Some flag/member to be set in urc_cciotopti()?